xlators/mgmt/glusterd/src/glusterd-utils.c

```diff
@@ -153,6 +153,28 @@ is_brick_mx_enabled(void)
     return ret ? _gf_false : enabled;
 }
 
+gf_boolean_t
+is_brick_graceful_cleanup_enabled(void)
+{
+    char *value = NULL;
+    int ret = 0;
+    gf_boolean_t enabled = _gf_false;
+    xlator_t *this = NULL;
+    glusterd_conf_t *priv = NULL;
+
+    this = THIS;
+
+    priv = this->private;
+
+    ret = dict_get_strn(priv->opts, GLUSTERD_BRICK_GRACEFUL_CLEANUP,
+                        SLEN(GLUSTERD_BRICK_GRACEFUL_CLEANUP), &value);
+
+    if (!ret)
+        ret = gf_string2boolean(value, &enabled);
```

Comment:
`ret` should be checked. It's ok if `dict_get_strn()` fails because it means the key is not specified and we are using the default value, but if it's present, it must be a boolean. If not, it's an error. I'm not sure if it's right to consider an error as the feature not being enabled.

Reply:
I feel, its better to just check for presence of key as `true`. just do `dict_get_sizen()`. Why one more conversion, and check.

Reply:
The conversion is required to take decision about enable/disable of the key. I think we should throw an error in case if it not boolean but currently we are not following it.